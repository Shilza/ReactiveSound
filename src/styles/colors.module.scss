
$colors: (
        'primary': #262728,
        'accent': #aaa,
        'secondary': #1d1e1f,
        'background': #262728
)