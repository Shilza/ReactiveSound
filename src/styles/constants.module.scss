$constants: (
        'maxWidth': 1000px
);
