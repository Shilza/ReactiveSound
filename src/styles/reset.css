
ul {
    list-style: none;
    padding: 0;
    margin: 0;
}

a {
    margin: 0;
    padding: 0;
    font-size: 100%;
    vertical-align: baseline;
    background: transparent;
    text-decoration: unset;
    font-weight: normal;
    color: currentColor;
}