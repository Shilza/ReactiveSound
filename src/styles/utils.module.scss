
@import 'colors.module';

@function color($key: 'primary') {
  @return map-get($colors, $key);
}