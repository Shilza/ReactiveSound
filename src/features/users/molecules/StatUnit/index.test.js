import React from 'react';
import {cleanup, render} from 'react-testing-library';
import {StatUnit} from "./index";
import styles from './styles.module.scss';

describe('<StatUnit/>', (() => {
    const count = 500;
    const active = true;
    const Children = <div data-testid='statUnitChildren'>Children</div>;
    const {container, queryByTestId} = render(
        <StatUnit count={count} active={active}>
            {Children}
        </StatUnit>
    );

    it('should renders correctly', () => {
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should have active class', () => {
        const children = queryByTestId('statUnitChildren');
        expect(children).toHaveAttribute('class', styles.labelActive);
    });

    it('should have label class', () => {
        cleanup();
        const {queryByTestId} = render(
            <StatUnit count={count} active={false}>
                {Children}
            </StatUnit>
        );

        const children = queryByTestId('statUnitChildren');
        expect(children).toHaveAttribute('class', styles.label);
    });
}));

