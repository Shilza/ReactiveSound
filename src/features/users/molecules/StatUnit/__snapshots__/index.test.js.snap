// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatUnit/> should renders correctly 1`] = `
<div
  class="unit"
>
  <div
    class="labelActive"
    data-testid="statUnitChildren"
  >
    Children
  </div>
  <span
    class="count"
  >
    0
  </span>
</div>
`;
